File: C++/Arrays/twosum.cpp
#include<iostream>
#include<vector>
#include <algorithm>
using namespace std;

// int twosum(vector<int>&nums, int target ){
    
    

// }


int main(){
    
    vector<int> nums = {3,5,7,9,10,4};
    int target = 12;
    int n =  nums.size();
    int left = 0;
    int right = n-1;
    sort(nums.begin(), nums.end());
    while (left<right)
    {
        /* code */
        int sum = nums[left] + nums[right];
        if (sum == target)
        {
            return {left};
        }
        else if (sum < target)
        {
            left++;
        }
        else
        {
            right--;
        }
        return {};
    }
    
    // twosum(nums,target);
}

File: JavaScript/twosum.js


function twosum(){
    let nums = [3,5,7,9,10,4];
    let n = nums.length;
    let target = 12;
    console.log(n);
    nums.sort(function(a,b){return a-b});
    let low = 0;
    let high = n-1;
    let sum = 0;
    while(low<high){
     sum = nums[low]+nums[high];
     if(nums[low] + nums[high] === target) console.log([low,high]);
     else if(sum>target) high--;
     else low++;
    }
    console.log([]);
}

twosum()